src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -232,6 +232,75 @@ public void markDirty()
         colony.markCitizensDirty();
     }
 
+    @Override
+    public boolean equals(final Object o)
+    {
+        if (this == o)
+        {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass())
+        {
+            return false;
+        }
+
+        final CitizenData data = (CitizenData) o;
+
+        if (id != data.id)
+        {
+            return false;
+        }
+        if (female != data.female)
+        {
+            return false;
+        }
+        if (strength != data.strength)
+        {
+            return false;
+        }
+        if (endurance != data.endurance)
+        {
+            return false;
+        }
+        if (charisma != data.charisma)
+        {
+            return false;
+        }
+        if (intelligence != data.intelligence)
+        {
+            return false;
+        }
+        if (dexterity != data.dexterity)
+        {
+            return false;
+        }
+        if (name != null ? !name.equals(data.name) : data.name != null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS864)